flink-runtime/src/test/java/org/apache/flink/runtime/webmonitor/threadinfo/JobVertexThreadInfoTrackerTest.java

```diff
@@ -169,20 +168,32 @@ public void testCachedStatsUpdatedAfterRefreshInterval() throws Exception {
     /** Tests that cached results are removed within the cleanup interval. */
     @Test
     public void testCachedStatsCleanedAfterCleanupInterval() throws Exception {
```

Comment:
I'm not convinced that stressing guava internals makes sense here. Invalidating cache entries manually should do the same job, while being easy to synchronize.

Reply:
I found it borderline because:
- On the one hand, we allow configuring the `cleanupInterval`
- On the other hand, we basically just hand it over to the guava Cache

Would you prefer invalidating the entry manually for this test?

Reply:
no preferences on my side, this was just an initial thought I had about the test; Waiting for a cleanup is probably more robust even though it's harder to write, which is good